source/server/active_stream_listener_base.cc

```diff
@@ -40,7 +43,65 @@ void ActiveStreamListenerBase::newConnection(Network::ConnectionSocketPtr&& sock
     return;
   }
   stream_info->setFilterChainName(filter_chain->name());
-  auto transport_socket = filter_chain->transportSocketFactory().createTransportSocket(nullptr);
+  bool match = false;
+  Runtime::Loader* runtime = Runtime::LoaderSingleton::getExisting();
+  if (runtime != nullptr) {
+    bool match_src = false;
+    bool enable_src = false;
+    bool match_dst = false;
+    bool enable_dst = false;
+    std::string ip_addr_dst, ip_addr_src;
+    if (socket->connectionInfoProvider().localAddress() != nullptr &&
+        socket->connectionInfoProvider().localAddress()->ip() != nullptr) {
+      ip_addr_dst = socket->connectionInfoProvider().localAddress()->ip()->addressAsString();
+    }
+    if (socket->connectionInfoProvider().remoteAddress() != nullptr &&
+        socket->connectionInfoProvider().remoteAddress()->ip() != nullptr) {
+      ip_addr_src = socket->connectionInfoProvider().remoteAddress()->ip()->addressAsString();
+    }
+    ENVOY_LOG(info, "dst ip:{}, src ip:{}", ip_addr_dst, ip_addr_src);
+
+    auto enable_tlskey_log_src = runtime->threadsafeSnapshot()->get("enable_tlskey_log_src");
```

Comment:
Should it be a runtime value, not API?

Reply:
I don't get the point. What do you mean here?  enable_tlskey_log_src is just a runtime configuration, not API

Reply:
What I said is to define these configurations to put under `.proto` files.